client/src/modules/services/services.js

```diff
@@ -2,115 +2,91 @@ angular.module('bhima.controllers')
 .controller('ServicesController', ServicesController);
 
 ServicesController.$inject = [
-  'ServiceService', '$translate', 'SessionService', 'ModalService', 'util', 'NotifyService'
+  'ServiceService', 'DepotService', 'SessionService',
+  'ModalService', 'NotifyService', 'uiGridConstants', '$state',
 ];
 
-function ServicesController(Services, $translate, SessionService, ModalService, util, Notify) {
+function ServicesController(Services, Depots, SessionService,
+  ModalService, Notify, uiGridConstants, $state) {
   var vm = this;
 
-  vm.choosen = {};
-  vm.state = 'default';
-  vm.view = 'default';
-  vm.projectId = SessionService.project.id;
-
-  vm.maxLength = util.maxTextLength;
-
   // bind methods
-  vm.create = create;
-  vm.update = update;
-  vm.cancel = cancel;
-  vm.submit = submit;
-  vm.del    = del;
-
-  // sets the module view state
-  function setState(state) {
-    vm.state = state;
+  vm.deleteService = deleteService;
+  vm.editService = editService;
+  vm.createService = createService;
+  vm.toggleFilter = toggleFilter;
+
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : true,
+    onRegisterApi     : onRegisterApiFn,
+    columnDefs : [
+      { field : 'name', displayName : 'FORM.LABELS.SERVICE', headerCellFilter : 'translate' },
+      { field : 'action',
+        width : 80,
+        displayName : '',
+        cellTemplate : '/modules/services/templates/action.tmpl.html',
+        enableSorting : false,
+        enableFiltering : false,
+      },
+    ],
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
   }
 
-  // fired on startup
-  function startup() {
-    // load Services
-    refreshServices();
-
-    // Cost Center Assignment - not yet implemented in 2.x
-    // Profit Center Assignment - not yet implemented in 2.x
-
-    setState('default');
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
   }
 
-  function cancel() {
-    setState('default');
-    vm.view = 'default';
-  }
+  function loadServices() {
+    vm.loading = true;
 
-  function create() {
-    vm.view = 'create';
-    vm.service = {};
-  }
-
-  // switch to update mode
-  // data is an object that contains all the information of a service
-  function update(data) {
-    setState('default');
-    vm.service= data;
-    vm.view = 'update';
+    Services.read(null, { full : 1 })
```

Comment:
We should make an issue, to make this service as other so that it will never require a full parameter